tools/ui-components/src/form-group/form-group.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useMemo, createContext } from 'react';
+import { FormGroupProps } from './types';
+
+export type FormContextProps = Pick<
+  FormGroupProps,
+  'controlId' | 'validationState'
+>;
+export const FormContext = createContext<FormContextProps>({});
+
+const FormGroup = React.forwardRef<HTMLDivElement, FormGroupProps>(
+  (
+    { className, validationState, controlId, as, ...props },
+    ref
+  ): JSX.Element => {
+    const controlIdContext = useMemo(() => controlId, [controlId]);
+    const validationStateContext = useMemo(
+      () => validationState,
+      [validationState]
+    );
+    const context = {
+      controlId: controlIdContext,
+      validationState: validationStateContext
+    };
```

Comment:
I have changed this from the previous code, the previous code is below, because it was an object which was compared by reference.

```ts
    const context = useMemo(
      () => ({ controlId, validationState }),
      [controlId, validationState]
    );
```

with it being an object compared by reference, it was no different that `new Set()`, but slightly worse because there will be more checkers for the same functionality.

The new code aims to compare them through as `string`, which compare them as values and reduce the workload on the functions.

Reply:
Will create a test repo for this, this can go in tho, refactoring it can be done in another PR, unless you think otherwise, which I am okay with

Reply:
Memo is actually bad here because the set isn't set in the formGroup, but in the validation logic, so I asking it to memo something, that should be memo in other place